src/dataflow/src/sink/kafka.rs

```diff
@@ -143,9 +143,30 @@ struct KafkaSink {
     encoder: Encoder,
     producer: ThreadedProducer<SinkProducerContext>,
     activator: timely::scheduling::Activator,
+    txn_timeout: Duration,
 }
 
 impl KafkaSink {
+    fn transition_on_error(&self, current_state: SendState, e: KafkaError) -> SendState {
+        error!(
+            "encountered error during kafka interaction. {} in state {:?} : {}",
```

Comment:
minor: if theres any way we could dump more info like partition id, offset, or which timestamp we were trying to commit that could be useful